spec/0.3-draft/spec.md

```diff
@@ -0,0 +1,1121 @@
+If this is your first read of the spec, skip over the details of the relation types and come back to them later.
+
+
+## Set Relation
+A set relation is a relation, which groups nodes together.
+
+- **URI**: `https://spec.canvasprotocol.org/rel/set/0.2`
+- **Name**: `@ocwg/rel/set`
+
+
+A set has the following properties in its `data` object:
+
+| Property  | JSON Type | OCIF Type   | Required     | Contents                  |
+|-----------|-----------|-------------|--------------|---------------------------|
+| `members` | `array`   | [ID](#id)[] | **required** | IDs of members of the set |
+
+- **members**: A list of IDs of nodes or relations that are part of the set.
+  Resources cannot be part of a set.
+
+**Example:** A set relation with three members:
+```json
+{
+  "type": "@ocwg/set",
+  "members": ["n1", "n2", "n3"]
+}
+```
+
+**Example:** A node using the set relation would look like this:
+```json
+{
+  "id": "nodeA",
+  "data": [
+    {
+      "type": "@ocwg/set",
+      "members": ["n1", "n2", "n3"]
+    }
+  ]
+}
+```
+
+JSON schema: [set-rel.json](core/set-rel.json)
+
+
+
+
+## Group Relation
+A group relation is a relation, which groups nodes together.
+It implies stronger semantics than a [set relation](#set-relation).
+
+- Name: `@ocwg/rel/group`
+- URI: `https://spec.canvasprotocol.org/rel/group/0.2`
+
+A group is modeled as a relation with a list of its members.
+
+| Property  | JSON Type | OCIF Type   | Required     | Contents                  |
+|-----------|-----------|-------------|--------------|---------------------------|
+| `members` | `array`   | [ID](#id)[] | **required** | IDs of members of the set |
+
+- **members**: A list of IDs of nodes (or relations, such as other groups) that are part of the group.
+
+**Semantics**
+- Groups can contain groups as members. Thus, all semantics apply recursively.
+- When a group is deleted, all members are deleted as well.
+- When a group is 'ungrouped,' the group itself is deleted, but its members remain.
+- When a member is deleted, it is removed from the group.
+
+
+JSON schema: [group-rel.json](core/group-rel.json)
+
+
+## Edge Relation
+An edge relates two elements (nodes and/or relation, mixing types is allowed).
+It supports directed and undirected bi-edges.
+
+- URI: `https://spec.canvasprotocol.org/rel/edge/0.2`
+- Name: `@ocwg/rel/edge`
+
+It has the following properties:
+
+| Property   | JSON Type | OCIF Type | Required     | Contents                  | Default |
+|------------|-----------|:----------|--------------|---------------------------|:--------|
+| `from`     | `string`  | [ID](#id) | **required** | ID of source element.     |         |
+| `to`       | `string`  | [ID](#id) | **required** | ID of target element.     |         |
+| `directed` | `boolean` |           | optional     | Is the edge directed?     | `true`  |
+| `rel`      | `string`  |           | optional     | Represented relation type |         |
```

Comment:
Add an optional pointer to a visual node that represents the relation visually.

Reply:
added